src/sql/workbench/parts/notebook/common/models/notebookModel.ts

```diff
@@ -486,12 +486,11 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	private updateActiveClientSession(clientSession: IClientSession) {
 		this.clearClientSessionListeners();
 		this._activeClientSession = clientSession;
-		this._clientSessionListeners.push(this._activeClientSession.kernelChanged(e => this._kernelChangedEmitter.fire(e)));
+		this._clientSessionListeners.add(this._activeClientSession.kernelChanged(e => this._kernelChangedEmitter.fire(e)));
```

Comment:
>_clientSessionListeners [](start = 7, length = 23)

It's weird that we only ever store one in here to begin with, but as to your above comment yeah I think it's reasonable to register the store as well to ensure the listener is cleaned up if this is disposed.

Reply:
Not super familiar with the code, so I didn't want to change existing functionality.